tests/unit/core/sagas/test_autocomplete.js

```diff
@@ -0,0 +1,79 @@
+import SagaTester from 'redux-saga-tester';
+
+import autocompleteReducer, {
+  autocompleteStart,
+  AUTOCOMPLETE_LOADED,
+} from 'core/reducers/autocomplete';
+import { CLEAR_ERROR } from 'core/constants';
+import * as api from 'core/api';
+import apiReducer from 'core/reducers/api';
+import authReducer from 'core/reducers/authentication';
+import autocompleteSaga from 'core/sagas/autocomplete';
+import { dispatchSignInActions } from 'tests/unit/amo/helpers';
+import { createStubErrorHandler } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockApi = sinon.mock(api);
+    const initialState = dispatchSignInActions().state;
+    sagaTester = new SagaTester({
+      initialState,
+      reducers: {
+        api: apiReducer,
+        auth: authReducer,
+        autocomplete: autocompleteReducer,
+      },
+    });
+    sagaTester.start(autocompleteSaga);
+  });
+
+  function _autocompleteStart(params) {
+    sagaTester.dispatch(autocompleteStart({
+      errorHandlerId: 'create-stub-error-handler-id',
+      ...params,
+    }));
+  }
+
+  it('calls the API for suggestions', async () => {
+    const results = sinon.stub();
+    const filters = { query: 'test' };
+
+    mockApi
+      .expects('autocomplete')
+      .once()
+      .returns(Promise.resolve({ results }));
+
+    _autocompleteStart({ filters });
+
+    // The saga should respond by dispatching the autocomplete loaded action.
+    await sagaTester.waitFor(AUTOCOMPLETE_LOADED);
+    mockApi.verify();
```

Comment:
Indeed. I must admit I have mimicked the whole search feature and this is missing there too 🙈